librz/arch/isa/xtensa/xtensa_il.c

```diff
@@ -0,0 +1,2178 @@
+}
+
+static RzAnalysisLiftedILOp op_ssxp(XtensaContext *ctx) {
+	return SEQ3(
+		SETL("vAddr", IREG(1)),
+		STOREW(VARL("vAddr"), V32(IREG(0))),
+		SETG(REGN(1), ADD(VARL("vAddr"), IREG(2))));
+}
+
+static RzAnalysisLiftedILOp op_sub(XtensaContext *ctx) {
+	return SETG(REGN(0), SUB(IREG(1), IREG(2)));
+}
+
+static RzAnalysisLiftedILOp op_subx2(XtensaContext *ctx) {
+	return SETG(REGN(0), SUB(SHIFTL0(IREG(1), U32(1)), IREG(2)));
+}
+
+static RzAnalysisLiftedILOp op_subx4(XtensaContext *ctx) {
+	return SETG(REGN(0), SUB(SHIFTL0(IREG(1), U32(2)), IREG(2)));
+}
+
+static RzAnalysisLiftedILOp op_subx8(XtensaContext *ctx) {
+	return SETG(REGN(0), SUB(SHIFTL0(IREG(1), U32(3)), IREG(2)));
+}
+
+// statusflags: VOI
+static RzAnalysisLiftedILOp op_sub_s(XtensaContext *ctx) {
+	return SEQ3(SETL("fres", FSUB(RZ_FLOAT_RMODE_RNA, FLOATV32(IREG(1)), FLOATV32(IREG(2)))),
+		SETG(REGN(0), UNSIGNED(64, F2BV(VARL("fres")))),
+		FSR_set(FSR_V | FSR_O | FSR_I, VARL("fres")));
+}
+
+static RzAnalysisLiftedILOp op_syscall(XtensaContext *ctx) {
+	return NOP();
+}
+
+// statusflags: VI
+static RzAnalysisLiftedILOp op_trunc_s(XtensaContext *ctx) {
+	return SEQ3(SETL("fres", FMUL(RZ_FLOAT_RMODE_RNA, FLOATV32(IREG(1)), F32(pow(2, IMM(2))))),
+		SETG(REGN(0), F2SINT(32, RZ_FLOAT_RMODE_RNA, VARL("fres"))),
+		FSR_set(FSR_V | FSR_I, VARL("fres")));
+}
+
+// statusflags: V
+static RzAnalysisLiftedILOp op_ueq_s(XtensaContext *ctx) {
+	return f_bool_op2(ctx, rz_il_op_new_feq);
+}
+
+// statusflags: I
+static RzAnalysisLiftedILOp op_ufloat_s(XtensaContext *ctx) {
+	return SEQ3(
+		SETL("fs", FMUL(RZ_FLOAT_RMODE_RNA, INT2F(RZ_FLOAT_IEEE754_BIN_32, RZ_FLOAT_RMODE_RNA, IREG(1)), F32(pow(2, -IMM(IMM(2)))))),
+		SETG(REGN(0), UNSIGNED(64, F2BV(VARL("fs")))),
+		FSR_set(FSR_I, VARL("fs")));
+}
+
+// statusflags: V
+static RzAnalysisLiftedILOp op_ule_s(XtensaContext *ctx) {
+	return f_bool_op2(ctx, rz_il_op_new_fle);
+}
+
+// statusflags: V
+static RzAnalysisLiftedILOp op_ult_s(XtensaContext *ctx) {
+	return f_bool_op2(ctx, rz_il_op_new_flt);
+}
+
+static RzAnalysisLiftedILOp op_umul_aa(XtensaContext *ctx) {
+	ut8 half = RRR_half(ctx);
+	return SEQ3(
+		SETG("m1", half & 0x1 ? HI16(IREG(0)) : LO16(IREG(0))),
+		SETG("m2", half & 0x2 ? HI16(IREG(1)) : LO16(IREG(1))),
+		ACC_set(MUL(V64(VARG("m1")), V64(VARG("m2")))));
+}
+
+// statusflags: V
+static RzAnalysisLiftedILOp op_un_s(XtensaContext *ctx) {
+	return SEQ2(SETG(REGN(0), OR(IS_FNAN(FLOATV32(IREG(1))), IS_FNAN(FLOATV32(IREG(2))))),
+		FSR_field_set(FSR_V, B2U32(OR(IS_FNAN(FLOATV32(IREG(1))), IS_FNAN(FLOATV32(IREG(2)))))));
+}
+
+// statusflags: VI
+static RzAnalysisLiftedILOp op_utrunc_s(XtensaContext *ctx) {
+	return SEQ3(SETL("fres", FMUL(RZ_FLOAT_RMODE_RNA, FLOATV32(IREG(1)), F32(pow(2, IMM(2))))),
+		SETG(REGN(0), F2INT(32, RZ_FLOAT_RMODE_RNA, VARL("fres"))),
+		FSR_set(FSR_V | FSR_I, VARL("fres")));
+}
+
+// TODO: interrupt
+static RzAnalysisLiftedILOp op_waiti(XtensaContext *ctx) {
+	return NOP();
+}
+
+// TODO: datatlb
```

Comment:
I think these todos aren't needed. There is no point of uplifting barrier instructions to RzIL, I think.